app/controllers/assessment/autograde.rb

```diff
@@ -71,12 +74,16 @@ def regradeBatch
     submission_ids.each do |submission_id|
       submission = @assessment.submissions.find_by_id(submission_id)
       if submission
-        job = autogradeSubmissions(@course, @assessment, [submission])
-        if job == -2 # no autograding properties for this assessment
-          redirect_to([@course, @assessment, :submissions]) && return
-        elsif job < 0 # autograding failed
-          failed_jobs += 1
-          failed_list += "#{@submission.filename}: autograding error.<br>"
+        begin
+          autogradeSubmissions(@course, @assessment, [submission])
+        rescue AssessmentAutogradeHelper::AutogradeError => e
+          if e.error_code == :missing_autograding_props
+            # no autograding properties for this assessment
+            redirect_to([@course, @assessment, :submissions]) && return
+          else # autograding failed
+            failed_jobs += 1
+            failed_list += "#{@submission.filename}: autograding error.<br>"
```

Comment:
Okay makes sense. But ideally we would want both API and web code to call the`AssessmentAutograde` module functions. Then in their individual caller code, do whatever error handling is specific to that platform. This + the helper introduces 3 layers of indirection. Does that make sense? We can discuss more tomorrow